File: MichiMark/Persistence/Repository/Settings/Action/ActionRepository.swift
import Foundation

protocol ActionRepository: Sendable {
    /// 論理削除されていない Action を全件取得
    func fetchAll() async throws -> [ActionDomain]

    /// 新規作成 or 上書き保存
    func save(_ action: ActionDomain) async throws

    /// 既存 Action の更新
    func update(_ action: ActionDomain) async throws
}

File: flutter/lib/repository/action_repository.dart
import '../domain/master/action/action_domain.dart';

/// アクションの永続化インターフェース
///
/// 削除は論理削除（isDeleted フラグ）で行うため、deleteメソッドは持たない。
/// 新規作成・更新は save で一本化する（upsert）。
abstract interface class ActionRepository {
  /// 論理削除されていないアクションを全件取得
  Future<List<ActionDomain>> fetchAll();

  /// アクションを保存（新規作成 / 上書き更新）
  Future<void> save(ActionDomain action);
}
